src/CVPresenter.ts

```diff
@@ -0,0 +1,129 @@
+import { IndexedFormula, NamedNode, Literal, Namespace } from "rdflib";
+import { ns, language } from "solid-ui";
+import { findImage } from "solid-ui/lib/widgets/buttons";
+import Node from "rdflib/src/node-internal";
+import { validateHTMLColorHex } from "validate-color";
+
+export interface Role {
+  startDate?: Literal,
+  endDate: Literal,
+  dates: string,
+  orgName: string,
+  roleText: string,
+  orgHomePage?: string
+}
+export interface CVPresentation { rolesByType: {
+    PastRole: Role[];
+    CurrentRole: Role[];
+    FutureRole: Role[];
+  };
+   skills: string[];
+  }
+
+const ORG = Namespace('http://www.w3.org/ns/org#');
+
+export const typesOfRole = ['PastRole', 'CurrentRole', 'FutureRole'];
+
+export function presentCV (
+  subject: NamedNode,
+  store: IndexedFormula
+): CVPresentation {
+  const profile = subject.doc()
+  const memberships = store.each(null, ORG('member'), subject, null)
+
+  function initialArray () {
+    const ar:Role[] = []
+    return ar
+  }
+
+ const rolesByType = { PastRole: [], CurrentRole: [], FutureRole: [] }
+  /*
+  for (const t of typesOfRole) {
+   rolesByType[t] = []
+  }
+  */
+  // const now = Literal.fromValue(new Date()) // toISOString
+  let endDate
+  for (const membership of memberships) {
+    let endDate, orgHomePage, orgNameGiven, publicIdName, roleName, publicId
+     if (store.holds(membership, ns.rdf('type'), ns.solid('PastRole'))) {
+       endDate = store.any(membership as any, ns.schema('endDate'))
+     }
+
+/* Just for the record this would not be a bad way to write this code in future
```

Comment:
I think it's good in this case, because it's giving us hints what we should do in the future.